### PR TITLE
[GPU] Pass concat unit tests on DG2

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Optimization checking logic update

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h [hunk 1]
```diff
@@ -47,6 +47,13 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
           _pd(pd),
           _prim(pd) { }
 
+    typed_primitive_onednn_impl(const typed_program_node<PType>& arg,
+                                const PrimDescType& pd)
+        : typed_primitive_impl<PType>({}, arg.can_be_optimized()?"undef":pd.impl_info_str()),
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h:52]: nit: please add space properly.

File: src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h [hunk 2]
```diff
@@ -47,6 +47,13 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
           _pd(pd),
           _prim(pd) { }
 
+    typed_primitive_onednn_impl(const typed_program_node<PType>& arg,
+                                const PrimDescType& pd)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h:51]: Do we need to accept `pd`? It is not used... And please add a comment to show that it is for "optimized" kernel.

File: src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h (hunk 1 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h]: Done.

File: src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h (hunk 2 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h]: Done. Added 'assert(arg.can_be_optimized());' in ctor to ensure it is optimized.

**Review by isanghao — APPROVED** [review]

**Pull request merged**